src/cs/Ssh.Tcp/Services/PortForwardingService.cs

```diff
@@ -538,17 +538,21 @@ public PortForwardingService(SshSession session)
 		}
 
 		request.ResponseTask = Task.FromResult(response ?? new SessionRequestFailureMessage());
-
-		// Add to the collection (and raise event) after sending the response,
-		// to ensure event-handlers can immediately open a channel.
-		if (response is PortForwardSuccessMessage portForwardResponse)
+		request.ResponseContinuation = () =>
```

Comment:
As the `response` is known already, we can check it before setting `ResponseContinuation`:
``` csharp
if (response is PortForwardSuccessMessage portForwardResponse)
{
    request.ResponseContinuation = () =>
    {
				var forwardedPort = new ForwardedPort(
					localPort: (int)portForwardResponse.Port,
					remotePort: portForwardRequest.Port == 0 ? null : (int)portForwardRequest.Port,
					isRemote: true);
				RemoteForwardedPorts.AddPort(forwardedPort);
    };
}
```

Reply:
Updated.